### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by nightscape**
This PR introduces [Mill](http://mill-build.com/) as build tool and uses it for easy cross-building against multiple Spark and Scala versions.
It builds on https://github.com/mayur2810/sope/pull/17 because ScalaTest 3.0.0 is not available for Scala 2.13.
I didn't touch the Maven build, so currently there is duplication in the two build definitions.
Not sure if you prefer or are tied to Maven, but Mill is a really nice and approachable alternative.
It replaces ~500 lines of XML with 50 lines of Scala.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
+name: Continuous Integration
```
Comment (mayur2810) [inline .github/workflows/ci.yml:1]: Hey @nightscape I was not actively looking at the project but thanks for the PR. Will it still work or need some changes?  I've not really used the Github actions so I'll need to figure this out.

File: build.sc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  def scalaTestVersion = "3.2.16"
+  def jacksonYamlVersion = "2.6.5"
+  def bigqueryConnectorVersion = "hadoop3-0.13.11"
+  def scoptVersion = "3.7.1"
+  def scoverageVersion = "2.0.10"
+
+  def pomSettings = PomSettings(
+    organization = "com.mayurb",
+    description = "Sope",
+    url = "https://github.com/mayurbhangale/sope",
```
Comment (mayur2810) [inline build.sc:21]: My github id is incorrect, will fix it.

Comment (nightscape) [reply build.sc]: I fixed it. Not sure where I got the wrong values from...

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nightscape) [reply .github/workflows/ci.yml:1]: Hi @mayur2810, the GitHub action is still working:
https://github.com/nightscape/sope/actions/runs/11456230896

**Pull request merged**